spalp_example/src/EventSubscriber/SpalpExampleConfigAlterSubscriber.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\spalp_example\EventSubscriber;
+
+use Drupal\spalp\Event\SpalpConfigAlterEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\spalp_example\SpalpExampleInterface;
+
+/**
+ * Class SpalpExampleConfigAlterSubscriber.
+ *
+ * @package Drupal\spalp_example\EventSubscriber
+ */
+class SpalpExampleConfigAlterSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[SpalpConfigAlterEvent::APP_CONFIG_ALTER] = 'doAppConfigAlter';
+
+    return $events;
+  }
+
+  /**
+   * React to app alter event to add additional config.
+   *
+   * @param \Drupal\spalp\Event\SpalpConfigAlterEvent $event
+   *   Spalp App Ids Alter Event.
+   */
+  public function doAppConfigAlter(SpalpConfigAlterEvent $event) {
+    if ($event->getAppId() === SpalpExampleInterface::APP_ID) {
+      $config = $event->getConfig();
+      $additional_config = [
```

Comment:
let's have a more meaningful example, e.g. 


```
if ($url == 'test.example.com') {
  // set an example API key or endpoint URL
}
```

this would also address https://www.drupal.org/project/spalp/issues/2996141